challenge.py

```diff
@@ -0,0 +1,156 @@
+import os
+import re
+
+
+List_calls = "calls.txt"
+List_files = "list_files.txt"
+result_stat = "stat.txt"
+
+#set flags to optimise computation files or call if not needed
+flag_file = 1
+flag_calls = 1
+def result(filename):
+  
+  #local variable declaration
+  deleted = 0 
+  added = 0 
+  region = 0 
+  calls = []
+  Files_array = []
+  
+  #Regular expression declaration 
+  regex_FileName = '^diff --git a.+[ ]b/(.+?)\n'
+  regex_Del = "^-(?!--).*"
+  regex_Add = "^\+(?!\+\+).*"
+  regex_region = "^@@ -\d+(?:,\d+)? \+\d+(?:,\d+)?\ @@[ ]?.*"
+  regex_functionCall = "([\w]+)(?=\()"
+  
+  #catch les nom des fonctions avec le regex, les mettres dans une liste, ensuite appliquer la fonction de l'occurence
+  regex_FileNamePattern = re.compile(regex_FileName)
+  regex_DelPattern = re.compile(regex_Del,re.DOTALL)
+  regex_AddPattern = re.compile(regex_Add,re.DOTALL)
+  regex_regionPattern = re.compile(regex_region)
+  regex_functionCallPattern = re.compile(regex_functionCall)
+  
+  #parcour each line of diff file
+  for line in open(filename):
+  #to optimize we did test to be sure we use a useful line
+   
+    #ignore lines starting with "+++" or "---" or "index" or "deleted" or "new"
+    if ( line.startswith("+++") or line.startswith("---") or line.startswith("index") or line.startswith("deleted") or line.startswith("new")):
```

Comment:
I somehow missed it during my first review, but how is comparing against all we don't want prior to comparing against all we want an optimization? You'll pay the cost of all the comparisons, especially since they are `OR`'d, this will actually slow down the whole process

Reply:
Okay, I agree :+1:  :)